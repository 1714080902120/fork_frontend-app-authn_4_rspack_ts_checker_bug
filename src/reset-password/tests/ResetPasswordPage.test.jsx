import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';
import * as auth from '@edx/frontend-platform/auth';

import ResetPasswordPage from '../ResetPasswordPage';

jest.mock('../data/selectors', () => jest.fn().mockImplementation(() => ({ resetPasswordSelector: () => ({}) })));
jest.mock('@edx/frontend-platform/auth');

const IntlResetPasswordPage = injectIntl(ResetPasswordPage);
const mockStore = configureStore();

describe('ResetPasswordPage', () => {
  let props = {};
  let store = {};

  const reduxWrapper = children => (
    <IntlProvider locale="en">
      <Provider store={store}>{children}</Provider>
    </IntlProvider>
  );

  beforeEach(() => {
    store = mockStore();
    props = {
      resetPassword: jest.fn(),
      status: null,
      token_status: 'pending',
      token: null,
      errors: null,
    };
  });

  it('should match reset password default section snapshot', () => {
    props = {
      ...props,
      token: 'token',
      token_status: 'valid',
    };
    const tree = renderer.create(reduxWrapper(<IntlResetPasswordPage {...props} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match invalid token message section snapshot', () => {
    props = {
      ...props,
      token_status: 'invalid',
    };
    const tree = renderer.create(reduxWrapper(<IntlResetPasswordPage {...props} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match successful reset message section snapshot', () => {
    props = {
      ...props,
      token_status: 'valid',
      status: 'success',
    };
    const tree = renderer.create(reduxWrapper(<IntlResetPasswordPage {...props} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match unsuccessful reset message section snapshot', () => {
    props = {
      ...props,
      token_status: 'valid',
      status: 'failure',
      errors: 'Password reset was unsuccessfull.',
    };
    const tree = renderer.create(reduxWrapper(<IntlResetPasswordPage {...props} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display invalid password message', async () => {
    const validationMessage = 'This password is too short. It must contain at least 8 characters. This password must contain at least 1 number.';
    const data = {
      validation_decisions: {
        password: validationMessage,
      },
    };
    props = {
      ...props,
      token_status: 'valid',
    };

    auth.getHttpClient = jest.fn(() => ({
      post: async () => ({
        data,
        catch: () => {},
      }),
    }));

    const resetPasswordPage = mount(reduxWrapper(<IntlResetPasswordPage {...props} />));
    await act(async () => {
      await resetPasswordPage.find('input#reset-password-input').simulate('blur');
    });
    resetPasswordPage.update();
    expect(resetPasswordPage.find('#reset-password-input-invalid-feedback').text()).toEqual(validationMessage);
  });
});
