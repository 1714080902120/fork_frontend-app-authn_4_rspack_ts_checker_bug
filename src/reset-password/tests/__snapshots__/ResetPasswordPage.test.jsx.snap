// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordPage should match invalid token message section snapshot 1`] = `
<div
  className="d-flex justify-content-center reset-password-container"
>
  <div
    className="d-flex flex-column"
    style={
      Object {
        "width": "400px",
      }
    }
  >
    <div
      className="form-group"
    >
      <div
        className="text-center mt-3"
      >
        <div
          className="alert d-flex align-items-start alert-danger mt-n2"
        >
          <div />
          <div>
            <h4
              style={
                Object {
                  "color": "#a0050e",
                }
              }
            >
              <span>
                Invalid Password Reset Link
              </span>
            </h4>
            <span>
              This password reset link is invalid. It may have been used already. To reset your password, go to the 
              <a
                href="/login"
                onClick={[Function]}
                target="_self"
              >
                <span>
                  sign-in
                </span>
              </a>
               page and select 
              <strong>
                Forgot Password
              </strong>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ResetPasswordPage should match pending reset message section snapshot 1`] = `
<div
  className="d-flex justify-content-center reset-password-container"
>
  <div
    className="d-flex flex-column"
    style={
      Object {
        "width": "450px",
      }
    }
  >
    <form
      className="m-4"
    >
      <div
        className="form-group"
      >
        <h3
          className="text-center mt-3"
        >
          Reset Your Password
        </h3>
        <p
          className="mb-4"
        >
          Enter and confirm your new password.
        </p>
        <div
          className="d-flex flex-column align-items-start"
        >
          <div
            className="form-group"
          >
            <label
              className="h6 mr-1"
              htmlFor="reset-password-input"
            >
              New Password
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="reset-password-input"
              name="new-password1"
              onBlur={[Function]}
              placeholder=""
              style={
                Object {
                  "width": "400px",
                }
              }
              type="password"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="h6 mr-1"
              htmlFor="confirm-password-input"
            >
              Confirm Password
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="confirm-password-input"
              name="new-password2"
              onChange={[Function]}
              placeholder=""
              style={
                Object {
                  "width": "400px",
                }
              }
              type="password"
              value=""
            />
            <strong
              className="invalid-feedback"
              id="confirm-password-input-invalid-feedback"
            >
              Passwords do not match.
            </strong>
          </div>
        </div>
      </div>
      <button
        aria-disabled={true}
        aria-live="assertive"
        className="pgn__stateful-btn pgn__stateful-btn-state-pending btn-primary submit disabled btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        <span
          className="d-flex align-items-center justify-content-center"
        >
          <span
            className="pgn__stateful-btn-icon"
          >
            <span
              aria-hidden={true}
              className="icon fa fa-spinner fa-spin"
              id="Icon1"
            />
          </span>
          Reset my password
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`ResetPasswordPage should match reset password default section snapshot 1`] = `
<div
  className="d-flex justify-content-center reset-password-container"
>
  <div
    className="d-flex flex-column"
    style={
      Object {
        "width": "450px",
      }
    }
  >
    <form
      className="m-4"
    >
      <div
        className="form-group"
      >
        <h3
          className="text-center mt-3"
        >
          Reset Your Password
        </h3>
        <p
          className="mb-4"
        >
          Enter and confirm your new password.
        </p>
        <div
          className="d-flex flex-column align-items-start"
        >
          <div
            className="form-group"
          >
            <label
              className="h6 mr-1"
              htmlFor="reset-password-input"
            >
              New Password
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="reset-password-input"
              name="new-password1"
              onBlur={[Function]}
              placeholder=""
              style={
                Object {
                  "width": "400px",
                }
              }
              type="password"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="h6 mr-1"
              htmlFor="confirm-password-input"
            >
              Confirm Password
            </label>
            <input
              aria-describedby=""
              className="form-control"
              id="confirm-password-input"
              name="new-password2"
              onChange={[Function]}
              placeholder=""
              style={
                Object {
                  "width": "400px",
                }
              }
              type="password"
              value=""
            />
            <strong
              className="invalid-feedback"
              id="confirm-password-input-invalid-feedback"
            >
              Passwords do not match.
            </strong>
          </div>
        </div>
      </div>
      <button
        aria-disabled={false}
        aria-live="assertive"
        className="pgn__stateful-btn pgn__stateful-btn-state-null btn-primary submit btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        <span
          className="d-flex align-items-center justify-content-center"
        >
          Reset my password
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`ResetPasswordPage should match successful reset message section snapshot 1`] = `
<div
  className="d-flex justify-content-center reset-password-container"
>
  <div
    className="d-flex flex-column"
    style={
      Object {
        "width": "400px",
      }
    }
  >
    <div
      className="form-group"
    >
      <div
        className="text-center mt-3"
      >
        <div
          className="fade alert alert-success mt-n2 show"
          role="alert"
        >
          <div
            className="alert-heading h4"
            style={
              Object {
                "color": "green",
              }
            }
          >
            <span>
              Password Reset Complete.
            </span>
          </div>
          <span>
            Your password has been reset. 
            <a
              className="font-weight-normal alert-link"
              href="/login"
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "color": "#0075b4",
                }
              }
            >
              <span>
                Sign-in to your account.
              </span>
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ResetPasswordPage should match unsuccessful reset message section snapshot 1`] = `
Array [
  <div
    className="d-flex justify-content-center reset-password-container"
  >
    <div
      className="d-flex flex-column"
      style={
        Object {
          "width": "400px",
        }
      }
    >
      <div
        className="form-group"
      >
        <div
          className="text-center mt-3"
        >
          <div
            className="fade alert alert-danger show"
            role="alert"
          >
            <div
              style={
                Object {
                  "color": "#a0050e",
                }
              }
            >
              <span>
                Password reset was unsuccessfull. 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="d-flex justify-content-center reset-password-container"
  >
    <div
      className="d-flex flex-column"
      style={
        Object {
          "width": "450px",
        }
      }
    >
      <form
        className="m-4"
      >
        <div
          className="form-group"
        >
          <h3
            className="text-center mt-3"
          >
            Reset Your Password
          </h3>
          <p
            className="mb-4"
          >
            Enter and confirm your new password.
          </p>
          <div
            className="d-flex flex-column align-items-start"
          >
            <div
              className="form-group"
            >
              <label
                className="h6 mr-1"
                htmlFor="reset-password-input"
              >
                New Password
              </label>
              <input
                aria-describedby=""
                className="form-control"
                id="reset-password-input"
                name="new-password1"
                onBlur={[Function]}
                placeholder=""
                style={
                  Object {
                    "width": "400px",
                  }
                }
                type="password"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                className="h6 mr-1"
                htmlFor="confirm-password-input"
              >
                Confirm Password
              </label>
              <input
                aria-describedby=""
                className="form-control"
                id="confirm-password-input"
                name="new-password2"
                onChange={[Function]}
                placeholder=""
                style={
                  Object {
                    "width": "400px",
                  }
                }
                type="password"
                value=""
              />
              <strong
                className="invalid-feedback"
                id="confirm-password-input-invalid-feedback"
              >
                Passwords do not match.
              </strong>
            </div>
          </div>
        </div>
        <button
          aria-disabled={false}
          aria-live="assertive"
          className="pgn__stateful-btn pgn__stateful-btn-state-failure btn-primary submit btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          <span
            className="d-flex align-items-center justify-content-center"
          >
            Reset my password
          </span>
        </button>
      </form>
    </div>
  </div>,
]
`;

exports[`ResetPasswordPage show spinner component during token validation 1`] = `
<div
  className="position-absolute no-inset"
>
  <div
    className="d-flex justify-content-center align-items-center full-vertical-height"
  >
    <div
      className="spinner-border text-primary"
    />
  </div>
</div>
`;
